report-generator/src/main/java/org/jacoco/core/internal/analysis/CachingClassAnalyzer.java

```diff
@@ -39,7 +37,17 @@ public void visitSource(String source, String debug) {
 	@Override
 	public MethodProbesVisitor visitMethod(final int access, final String name,
 										   final String desc, final String signature, final String[] exceptions) {
-		return new CachingMethodAnalyzer(classCoverageLookup, Filters.ALL, this);
+		final CachingInstructionsBuilder builder = new CachingInstructionsBuilder(classCoverageLookup);
+
+		return new MethodAnalyzer(builder) {
```

Comment:
please comment why overriding that one method is necessary. This might be confusing for the "casual" reader ;-)